src/test/java/generator/DecisionGeneratorTest.java

```diff
@@ -0,0 +1,72 @@
+package generator;
+
+import org.junit.Assert;
+import org.junit.Test;
+import org.mockito.stubbing.Answer;
+import project.*;
+
+import java.util.*;
+
+import static org.mockito.ArgumentMatchers.anyMap;
+import static org.mockito.Mockito.*;
+
+public class DecisionGeneratorTest {
+    private Map<Model.Side, TreeMap<Integer, Lane>> getExampleLaneMap(int leftLanes, int rightLanes){
+        Map<Model.Side, TreeMap<Integer, Lane>> lanesMap = new HashMap<>();
+        TreeMap<Integer, Lane> leftTreeMap = new TreeMap<>();
+        for(int i=1; i<=leftLanes; i++) {
+            leftTreeMap.put(i, null);
+        }
+        TreeMap<Integer, Lane> rightTreeMap = new TreeMap<>();
+        for(int i=1; i<=rightLanes; i++) {
+            rightTreeMap.put(i, null);
+        }
+        TreeMap<Integer, Lane> centerTreeMap = new TreeMap<>();
+        centerTreeMap.put(0, null);
+        lanesMap.put(Model.Side.LEFT, leftTreeMap);
+        lanesMap.put(Model.Side.RIGHT, rightTreeMap);
+        lanesMap.put(Model.Side.CENTER, centerTreeMap);
+        return lanesMap;
+    }
+
+    @Test
+    public void generateActionByDecisionMapTest() {
+        Map<Decision, Action> actionByDecision = new HashMap<>();
+        String baseIRI = "anything";
+        MyFactory factoryMock = mock(MyFactory.class);
+        Change_lane changeLaneMock = mock(Change_lane.class);
+        Model modelMock = mock(Model.class);
+        Scenario scenarioMock = mock(Scenario.class);
+
+        doNothing().when(changeLaneMock).addLane_change_by(anyInt());
+        when(factoryMock.createDecision(anyString())).thenAnswer((Answer<Decision>) invocationOnMock -> mock(Decision.class));
+        when(factoryMock.createFollow(anyString())).thenReturn(mock(Follow.class));
+        when(factoryMock.createStop(anyString())).thenReturn(mock(Stop.class));
+        when(factoryMock.createTurn_left(anyString())).thenReturn(mock(Turn_left.class));
+        when(factoryMock.createTurn_right(anyString())).thenReturn(mock(Turn_right.class));
+        when(factoryMock.createChange_lane(anyString())).thenReturn(changeLaneMock);
+        doNothing().when(scenarioMock).addHas_decision(any());
+        when(modelMock.getScenario()).thenReturn(scenarioMock);
+        doNothing().when(modelMock).setActionByDecision(anyMap());
```

Comment:
kolejny problem z ontologią, kolejny powód, żeby do factory jak najmniej się odnosić

Reply:
Znaczy ten kod jako test jest ok, tylko prędzej kod DecisionGeneratora jest do zmiany? No generalnie nie wiem czy możemy tego uniknąć bo to jest jeszcze pakiet generator i jakby nie było generujemy dalej scenariusz.

Reply:
Tutaj bardziej narzekam na życie, niż konstruktywną krytykę piszę ;d